taroscript/send.go

```diff
@@ -354,6 +359,132 @@ func CreateOutputCommitments(inputCommitment *commitment.TaroCommitment,
+		if sortedCopy[i].AnchorOutputIndex != uint32(i) {
+			taroOnlySpend = false
+			break
+		}
+	}
+
+	return taroOnlySpend, nil
+}
+
+// CreateAnchorTx creates a template BTC anchor TX with dummy outputs.
+func CreateAnchorTx(outputs []*taropsbt.VOutput) (*psbt.Packet, error) {
+	// Check if our outputs are valid, and if we will need to add extra
+	// outputs to fill in the gaps between outputs.
+	taroOnlySpend, err := AreValidAnchorOutputIndexes(outputs)
+	if err != nil {
+		return nil, err
+	}
+
+	// Calculate the number of outputs we need for our template TX.
+	maxOutputIndex := uint32(len(outputs))
+
+	// If there is a gap in our outputs, we need to find the
+	// largest output index to properly size our template TX.
+	if !taroOnlySpend {
+		maxOutputIndex = 0
+		for _, out := range outputs {
+			if out.AnchorOutputIndex > maxOutputIndex {
+				maxOutputIndex = out.AnchorOutputIndex
+			}
+		}
+
+		// Output indexes are 0-indexed, so we need to increment this
+		// to account for the 0th output.
+		maxOutputIndex++
+	}
+
+	txTemplate := wire.NewMsgTx(2)
+	for i := uint32(0); i < maxOutputIndex; i++ {
+		txTemplate.AddTxOut(createDummyOutput())
+	}
+
+	spendPkt, err := psbt.NewFromUnsignedTx(txTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("unable to make psbt packet: %w", err)
+	}
+
+	return spendPkt, nil
+}
+
+// UpdateTaprootOutputKeys updates a PSBT with outputs embedding TaroCommitments
+// involved in an asset send. The sender must attach the Bitcoin input holding
+// the corresponding Taro input asset to this PSBT before finalizing the TX.
+// Locators MUST be checked beforehand.
+func UpdateTaprootOutputKeys(btcPacket *psbt.Packet,
+	outputs []*taropsbt.VOutput,
+	outputCommitments []*commitment.TaroCommitment) error {
+
+	// Add the commitment outputs to the BTC level PSBT now.
+	for idx := range outputs {
+		vOut := outputs[idx]
+		outputCommitment := outputCommitments[idx]
+
+		// The external output index cannot be out of bounds of the
+		// actual TX outputs. This should be checked earlier and is just
+		// a final safeguard here.
+		if vOut.AnchorOutputIndex >= uint32(len(btcPacket.Outputs)) {
+			return ErrInvalidOutputIndexes
+		}
+
+		btcOut := btcPacket.Outputs[vOut.AnchorOutputIndex]
+		internalKey, err := schnorr.ParsePubKey(
+			btcOut.TaprootInternalKey,
+		)
+		if err != nil {
+			return err
+		}
+
+		// The commitment must be defined at this point.
+		//
+		// TODO(guggero): Merge multiple Taro level commitments that use
+		// the same external output index.
+		if outputCommitment == nil {
+			return ErrMissingTaroCommitment
+		}
+
+		// Create the scripts corresponding to the receiver's
+		// TaroCommitment.
+		//
+		// NOTE: We currently default to the Taro commitment having no
+		// sibling in the Tapscript tree. Any sibling would need to be
+		// checked to verify that it is not also a Taro commitment.
+		script, err := PayToAddrScript(
+			*internalKey, nil, *outputCommitment,
```

Comment:
I think we also already pass the sibling along side the outputs? So then can just pass that here, knowing it'll be `nil` in most places we care about.

Reply:
No, we only pass around the taproot sibling for the _input_ we spend. So this would need to be defined on the BTC transaction level. I'll take a look at that when I do the PR that allows external signing of the BTC PSBT stuff (which will also be the PR that makes sure everything works with those Taproot siblings).